SABnzbd.py

```diff
@@ -1539,6 +1539,23 @@ def main():
 
         # 30 sec polling tasks
         if not timer % 10:
+            servers = config.get_servers()
+            for srv in config.get_servers():
+                server = servers[srv]
+                if server.expire_date():
+                    if time.time() > time.mktime(time.strptime(server.expire_date(), "%Y-%m-%d")):
```

Comment:
I doubt there is a need to check this every 30 seconds. The date check should be done at program start and using the scheduler at midnight. We already do that for the general quoata check.

Although I wonder what would be the point of such a notification? Once your server expires you'll get notified anyway by the errors/warnings that the server is ignored due to errors. Usually the newsserver reports that the quota is exceeded and SABnzbd will report that.

Reply:
That was just for testing, I forgot to change it. I intended it to be on the 5 minute mark, or create an even less regular one. It seemed simpler to keep the two tests together and only at one place in the code, and because it's done rarely the performance impact should be minimal.

Both are intended as soft early warnings, so you can renew or get a different subscription before they expire. If the main, usually time based, subscription expires you can end up using a lot of the data from the block based backup servers. On a backup server where you have 500 GB, you can add a warning when has used 400 GB so you can buy another block.

Reply:
This spot just in `SABnzbd.py ` isn't intended for these checks. It should only be used for program-essential stuff.
I think we need to move this to the scheduler.